spec/graphql/resolvers/customers/subscriptions_resolver_spec.rb

```diff
@@ -0,0 +1,178 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Resolvers::Customers::SubscriptionsResolver, type: :graphql do
+  let(:required_permission) { "customers:view" }
+  let(:query) do
+    <<~GQL
+      query($customerId: ID!) {
+        customer(id: $customerId) {
+          subscriptions(status: [#{status_filter}]) {
+            id
+            status
+            startedAt
+            plan {
+              code
+            }
+          }
+        }
+      }
+    GQL
+  end
+
+  let(:membership) { create(:membership) }
+  let(:organization) { membership.organization }
+  let(:customer) { create(:customer, organization:) }
+  let(:plan) { create(:plan, organization:) }
+  let(:status_filter) { nil }
+
+  before do
+    customer
+  end
+
+  it_behaves_like "requires permission", "customers:view"
+
+  describe "when no status filter is provided" do
+    let!(:active_subscription) { create(:subscription, customer:, plan:, status: "active") }
+    let!(:terminated_subscription) { create(:subscription, customer:, plan:, status: "terminated") }
+    let!(:pending_subscription) { create(:subscription, customer:, plan:, status: "pending", started_at: 1.day.from_now) }
+    let!(:downgraded_pending) { create(:subscription, customer:, plan:, status: "pending") }
+    let(:active_from_downgrade) { create(:subscription, customer:, plan:, status: "active", previous_subscription: downgraded_pending) }
```

Comment:
```suggestion
    let!(:downgraded_pending) { create(:subscription, customer:, plan:, status: "pending", previous_subscription: active_from_downgrade) }
    let(:active_from_downgrade) { create(:subscription, customer:, plan:, status: "active") }
```

or use `next_subscriptions` instead...